docs/how-to-write-a-contract/how-to-write-a-contract.md

```diff
@@ -406,6 +406,10 @@ let aab: FixedArray<bigint, N> = [1n, 2n]
 let abb: FixedArray<FixedArray<bigint, 2>, 3> = [[1n, 3n], [1n, 3n], [1n, 3n]]
 ```
 
+:::note
+Customized type and Fixed Array are *pass by reference*. When passing them to a contract method, an alias or reference to the actual parameter is passed, which means the method will access the actual parameter.
```

Comment:
Whereas in Javascript objects and arrays are passed by reference, in sCrypt all parameters are passed by value, which means you cannot pass an array or user-defined type to two different function parameters. for example:

```ts
// valid example
// invalid exampe

```